packages/itwin/tree-widget/src/components/trees/stateless/models-tree/ModelsTreeDefinition.ts

```diff
@@ -226,22 +258,40 @@ export class ModelsTreeDefinition implements HierarchyDefinition {
                         WHEN (
                           json_extract([partition].JsonProperties, '$.PhysicalPartition.Model.Content') IS NOT NULL
                           OR json_extract([partition].JsonProperties, '$.GraphicalPartition3d.Model.Content') IS NOT NULL
+                          ${
+                            this._hierarchyConfig.showEmptyModels
+                              ? ""
+                              : `AND EXISTS (
+                                  SELECT 1
+                                  FROM ${this._hierarchyConfig.elementClassSpecification} element
+                                  WHERE element.Model.Id = m.ECInstanceId
+                                )`
+                          }
```

Comment:
Is this necessary? `childModelIds` depend on the value of `showEmptyModels`, so it looks like with `showEmptyModels == false` we won't even select a model if it has no elements.

Reply:
No, removed.